### PR TITLE
Feat add forum controller

### PR DESCRIPTION
**Pull request by MichaelSlatteryDev**
For issue #133 

Still need to add forums permissions and tests too. After that forums and threads should be all done.

Some questions I had while implementing this. I think app side will need to send the parent forum id in the request when creating a forum. I can then get the parentList from that and make the new forum based on that. I don't think it's possible to get the parent information otherwise?

Also need to agree on what default forum permissions should be for each user role.

### CODE REVIEW

File: src/api/rest/forum/v1/controller/forum.controller.ts
Comment (Xazin) [inline src/api/rest/forum/v1/controller/forum.controller.ts:41]:
```suggestion
    async getForumHierarchy(@Param('forumId') id: string): Promise<Forum[]> {
```

File: src/modules/forum/cqrs/command/create-forum.command.ts
Comment (Xazin) [inline src/modules/forum/cqrs/command/create-forum.command.ts:28]:
```suggestion

```

File: src/infrastructure/forum/dto/forum.dto.ts
Comment (Xazin) [inline src/infrastructure/forum/dto/forum.dto.ts:57]:
```suggestion
    readonly parentId?: string;
```

**Review by Xazin — APPROVED** [review]

**Comment by MichaelSlatteryDev** [comment]
Even though this is all passing now once this is merged in development will break. We need a default forum from now on to make crowdactions. Please don't merge this in until it's confirmed to be ok

**Comment by Xazin** [comment]
This reminds me we need to fix migrations not working in CICD (I temporarily took it out when migrating to azure).

We can merge this though right? If not forum with the default crowdaction boolean, then creating a crowdaction simply does not create a thread right? (At least that is how I believe it should work)

**Comment by MichaelSlatteryDev** [comment]
> This reminds me we need to fix migrations not working in CICD (I temporarily took it out when migrating to azure).
> 
> We can merge this though right? If not forum with the default crowdaction boolean, then creating a crowdaction simply does not create a thread right? (At least that is how I believe it should work)

Now it will. Can you make an issue with the details about migrations? We'll also need to migrate user accounts that were made before this change to have thread and post counts. Right now I just default to 0 if it doesn't exist.

**Pull request merged**
